src/hooks/useData.js

```diff
@@ -8,7 +8,11 @@ export const initDataState = (data) => {
   if (dataState) return false;
   dataState = atom({
     key: "data",
-    default: data || {},
+    default: {
+      currency: { symbol: '€', code: 'EUR' },
```

Comment:
I needed a way to set these default values much earlier - setting initial values in components can easily cause re-rendering loops!

What do you think about creating an initial config with all of the basic entries? Would that simplify code in other places too?

Reply:
I don't get why you removed "data ||" ? is this what caused the re-render?

Reply:
No - the re-renders came from setData call in components used by components that also subscribed to changes in the recoil data.

I'm not sure I understand your question though - replacing the || operator with the spread operator let's us set default values that can be overridden by whatever is in data.

Is that what you were asking?

Reply:
there is a change in functionality with your modification, before, the default was whatever was set in the data, now it's always euro

I'm not sure anymore (and the commit doesn't show the context and I'm lazy ;) where that data came from, but if it isn't a problem, can you put it back?

Reply:
ah, the diff didn't show the last line with "...data", now it makes sense and clear